nx/include/switch/kernel/semaphore.h

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 Kevoot
+
+#ifndef __SEMAPHORE_H
+#define __SEMAPHORE_H
+
+#include "mutex.h"
+
+#define SEM_DOWN false
+#define SEM_UP   true
+
+#define EBUSY 1
+
+typedef struct sem_t
+{
+    bool flag;
+    Mutex mutex;
+} sem_t;
+
+void sem_init(sem_t *);
```

Comment:
According to naming convention this should be `semaphoreInit` (same for all the other ones)